core/packages/PackageDB.cc

```diff
@@ -168,6 +174,24 @@ const std::string_view PackageDB::errorHint() const {
     return errorHint_;
 }
 
+const std::string_view PackageDB::getLayerName(uint8_t layerRank) const {
+    ENFORCE(layerRank < layerNames_.size());
+    return layerNames_.at(layerRank);
+}
+
+const uint8_t PackageDB::getLayerRank(std::string layerName) const {
+    for (std::size_t i = 0; i < layerNames_.size(); i++) {
+        if (layerNames_.at(i).compare(layerName) == 0) {
```

Comment:
The implementation currently compares full string equality for names.

What do you think about entering the layer names with `gs.enterNameUtf8`, and storing either `UnorderedMap<NameRef, uint8_t>` or `vector<NameRef>` instead?

Reply:
An example of something similar is `AutoloaderConfig::enterConfig`